### DISCUSSION TITLE
Editing Validation in igx-grid

### DISCUSSION THREAD

**Discussion by kevalgw** [IgniteUI/igniteui-angular#12154] [Q&A] [answered]
Hi,

In my igx-grid there are some editable fields & I want  to restrict user from entering more than 7 characters in that respective field. 
Is there any way I can achieve it? 
Please respond.

![image](https://user-images.githubusercontent.com/101707048/193517308-a30257f3-d661-43f2-ae39-9fc56ea78655.png)

**Comment by ddaribo** [comment, marked as answer]
Hi @kevalgw,

Yes, there exist multiple approaches to restrict the cell’s text length. 
For example, for a text field, one option would be to leverage the `igxMask` directive, about which you could read more in our documentation [here](https://www.infragistics.com/products/ignite-ui-angular/angular/components/mask).
An [IgxInputGroup](https://www.infragistics.com/products/ignite-ui-angular/angular/components/input-group) having the `igxMask` directive could be placed in the target field’s [cell editing template](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/grid#cell-editing-template) as follows:  

```
<igx-column
    field="ProductName"
    header="Product Name"
    dataType="string"
    [editable]="true">
    <ng-template igxCellEditor let-cell="cell" let-value>
      <igx-input-group>
        <input
          igxInput
          type="text"
          [igxMask]="'CCCCCCC'"
          [igxTextSelection]="true"
          [igxFocus]="true"
          [(ngModel)]="cell.editValue"/>
      </igx-input-group>
    </ng-template>
  </igx-column>
```

The above mask specifies an input sequence of seven arbitrary keyboard characters. This restricts the length to 7 and provides prompt chars in addition.

For other field types, such as number, a possible approach would be to again create a custom cell editing template with the `igxInput` directive, however, in this case I can suggest handling the native `input` event of the control:

```
  <igx-column
    field="UnitPrice"
    header="Price"
    dataType="number"
    [editable]="true">
    <ng-template igxCellEditor let-cell="cell" let-value>
      <igx-input-group>
        <input
          igxInput
          type="number"
          [igxFocus]="true"
          [(ngModel)]="cell.editValue"
          (input)="handleUnitPriceInput($event, cell)"/>
      </igx-input-group>
    </ng-template>
  </igx-column>
```

And execute custom logic disallowing to type more than X characters, for instance:

```
  public handleUnitPriceInput(event: any, cell: IgxGridCell) {
    if (event.target.tagName.toLowerCase() !== 'input') {
      return;
    }
    let inputText = event.target.value;
    if (event.target.value.toString().length > 7) {
      event.target.value = inputText.slice(0, -1);
      cell.editValue = event.target.value;
    }
  }
```

Please, note that passing the `cell` object and modifying its `editValue` would be essential for this to work properly.

[Here](https://stackblitz.com/edit/igx-grid-restrict-cell-text-length?file=src/app/grid/grid-editing-events/grid-editing-events.component.html) you can find a sample demonstrating both approaches for the “ProductName” string field and for the “Price” number field.

**Reply by kevalgw** [reply]
Hi @ddaribo 

Thanks , appreciate your support. That code worked For me
